pkg/await/await.go

```diff
@@ -305,27 +315,14 @@ func Deletion(
 		return err
 	}
 
-	// Wait until delete resolves as success or error. Note that the conditional is set up to log only
-	// if we don't have an entry for the resource type; in the event that we do, but the await logic
-	// is blank, simply do nothing instead of logging.
-	var waitErr error
-	id := fmt.Sprintf("%s/%s", gvk.GroupVersion().String(), gvk.Kind)
-	if awaiter, exists := awaiters[id]; exists && awaiter.awaitDeletion != nil {
-		waitErr = awaiter.awaitDeletion(ctx, clientForResource, name)
-	} else {
-		objectMissing := func(d *unstructured.Unstructured, err error) error {
-			if is404(err) {
-				return nil
-			} else if err != nil {
-				glog.V(3).Infof("Received error deleting object '%s': %#v", d.GetName(), err)
-				return err
-			}
-			_ = host.LogStatus(ctx, diag.Info, urn, fmt.Sprintf("Waiting for deletion of %s '%s'", id, name))
-			return watcher.RetryableError(fmt.Errorf("Object '%s' still exists", name))
+	for event := range ch {
```

Comment:
In addition to monitoring events from the watch API, we probably also want to watch for `<-ctx.Done()` (as we do in [the `watcher` package](https://github.com/pulumi/pulumi-kubernetes/blob/2efb314c64218897bb88895bc468e7c231c8ee88/pkg/watcher/watcher.go#L129)). This allows users to ctrl-c in the middle of a delete. (See comment below about the error channel for an idea of what this means.)